packages/suite/src/components/suite/modals/Passphrase/components/PassphraseTypeCard/index.tsx

```diff
@@ -179,21 +179,26 @@ const PassphraseTypeCard = (props: Props) => {
 
     const isTooLong = countBytesInString(value) > MAX_LENGTH.PASSPHRASE;
 
-    const submit = (value: string, passphraseOnDevice?: boolean) => {
-        if (!enabled) return;
-        props.onSubmit(value, passphraseOnDevice);
-    };
+    const { onSubmit } = props;
+    const submit = useCallback(
+        (value: string, passphraseOnDevice?: boolean) => {
+            if (!enabled) return;
+            onSubmit(value, passphraseOnDevice);
+        },
+        [enabled, onSubmit],
+    );
 
-    if (enterPressed) {
-        // Trigger submit on pressing Enter in case of single col modal (creating/confirming hidden wallet)
-        // In case of two-col modal (selecting between standard and hidden wallet)
-        // only the hidden wallet part handle the enter press.
-        if (props.singleColModal || props.type === 'hidden') {
-            if (!isTooLong) {
-                submit(value);
-            }
+    const shouldSubmit =
+        enterPressed && (props.singleColModal || props.type === 'hidden') && !isTooLong;
+
+    // Trigger submit on pressing Enter in case of single col modal (creating/confirming hidden wallet)
+    // In case of two-col modal (selecting between standard and hidden wallet)
+    // only the hidden wallet part handle the enter press.
+    useEffect(() => {
+        if (shouldSubmit) {
```

Comment:
nit: Using `import { useHotkeys } from 'react-hotkeys-hook';` would be more clear

Reply:
https://github.com/trezor/trezor-suite/pull/4491/commits/af508207d6b01c069282832ce03329c66417e553